holoviews/core/data/pandas.py

```diff
@@ -332,12 +359,21 @@ def sort(cls, dataset, by=None, reverse=False):
     def select(cls, dataset, selection_mask=None, **selection):
         df = dataset.data
         if selection_mask is None:
-            selection_mask = cls.select_mask(dataset, selection)
+            indexes = {
+                idx or "index": v
+                for idx in df.index.names
+                if isinstance((v := selection.get(idx)), slice)
+            } if not isinstance(df.index, pd.MultiIndex) else {}
+            if selection.keys() - indexes.keys():
+                selection_mask = cls.select_mask(dataset, selection)
+            else:
+                for v in indexes.values():
+                    df = df[v]
```

Comment:
Wouldn't it be better to build up a full `indexes` tuple and then use `df.loc[indexes, :]`? Also I think you may have to transform two-tuples into slices here, since `.select(index=(0, 3))` should be treated like `.select(index=slice(0, 3))` (maybe that's handled elsewhere though).

Reply:
> Wouldn't it be better to build up a full indexes tuple and then use df.loc[indexes, :]

This fast-path only works for a single Index. Without transforming it into an array, I couldn't figure out an easy way to get it to work with `pd.MultiIndex`. So, even if it is a for-loop, it is only a single iteration. If I recall correctly, I did some tests and saw better speeds with `df[...]` over `df.loc[...]`. 

> Also I think you may have to transform two-tuples into slices here

It is not. It could also be for a tuple size of 3, right?

Reply:
I have pushed for the tuples.

Reply:
>Without transforming it into an array, I couldn't figure out an easy way to get it to work with pd.MultiIndex

Not totally following this bit. Want me to take a quick look at this?